util/types.go

```diff
@@ -525,6 +526,7 @@ func (ub *UBlock) SerializeCompactSize() int {
 // rebuilt from the block data.  Note that this leaves the blockhash
```

Comment:
The blockhash is not yet being restored right? at least i can't find it in any of the commits.
This also assumes a height to blockhash index on the client side exists, might not be worth it if the saving aren't huge.